new-integration-tests/src/test/java/oracle/weblogic/kubernetes/assertions/impl/Kubernetes.java

```diff
@@ -154,27 +153,30 @@ public static boolean isPodReady(String namespace, String domainUid, String podN
   }
 
   /**
-   * Checks if a pod exists in a given namespace and in Terminating state.
+   * Check if a pod exists in a given namespace and is terminating.
    * @param namespace in which to check for the pod
    * @param domainUid the label the pod is decorated with
    * @param podName name of the pod to check for
-   * @return true if pod is in Terminating state otherwise false
+   * @return true if pod is terminating otherwise false
    * @throws ApiException when there is error in querying the cluster
    */
-  public static boolean isPodTerminating(String namespace, String domainUid, String podName) throws ApiException {
```

Comment:
As an alternative, you can check that the pod's metadata.deletionTimestamp is set. If this field has a value, then the pod has been marked for deletion. It will either then be in it's terminating phase or soon will be.

Reply:
Yes the implementation checks for the existence of pod deletion time stamp.

Reply:
Yes, the new implementation uses pod deletion timestamp.